sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2404,22 +2420,33 @@ static ur_result_t SetKernelParamsAndLaunch(
         DeviceImageImpl ? DeviceImageImpl->get_spec_const_blob_ref() : Empty);
   }
 
+  // just a performance optimization - avoid heap allocations
+  static thread_local std::vector<ur_exp_kernel_arg_properties_t> UrArgs;
```

Comment:
This line introduce another perf regression. Since we are using TLS here there are a lot of calls to the `__tls_get_addr` on each call to this function. Actually on each access to the `UrArgs`.